service_container/service_decoration.rst

```diff
@@ -313,4 +313,117 @@ The generated code will be the following::
 
     $this->services[Foo::class] = new Baz(new Bar(new Foo()));
 
+Control the Behavior When the Decorated Service Does Not Exist
+--------------------------------------------------------------
+
+When you decorate a service that doesn't exists, the ``decoration_on_invalid``
+option allows you to choose the behavior to adopt.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        Foo: ~
+
+        Bar:
+            public: false
+            decorates: Foo
+            decoration_on_invalid: ignore
+            arguments: ['@Bar.inner']
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="Foo"/>
+
+                <service id="Bar" decorates="Foo" decoration-on-invalid="ignore">
+                    <argument type="service" id="Bar.inner"/>
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+        use Symfony\Component\DependencyInjection\ContainerInterface;
+
+        return function(ContainerConfigurator $configurator) {
+            $services = $configurator->services();
+
+            $services->set(Foo::class);
+
+            $services->set(Bar::class)
+                ->decorate(Foo::class, null, 0, ContainerInterface::IGNORE_ON_INVALID_REFERENCE)
+                ->args([ref(Bar::class.'.inner')]);
+
+            $services->set(Baz::class)
+                ->private()
+                ->decorate(Foo::class, null, 1)
+                ->args([ref(Baz::class.'.inner')]);
+        };
+
+Three different behaviors are available:
+
+* ``exception``: When used, a ``ServiceNotFoundException``
+  will be thrown telling that decorator's dependency is missing. (default)
+* ``ignore``: Using that behavior, the container will remove either decorator
```

Comment:
You can remove the "Using that behavior, "

Also it will remove the decorator only, because the decorated is already missing :)